mlflow/artifacts/__init__.py

```diff
@@ -133,7 +133,7 @@ def list_artifacts(
         return get_artifact_repository(artifact_uri=root_uri).list_artifacts(artifact_path)
 
     # Use `runs:/<run_id>/<artifact_path>` to list both run and model (if exists) artifacts
-    if run_id and artifact_path:
+    if run_id and artifact_path and tracking_uri is None:
         return get_artifact_repository(artifact_uri=f"runs:/{run_id}").list_artifacts(artifact_path)
```

Comment:
This can be a follow-up work. We might want to add `tracking_uri` to artifact repos. Some repos such as `RunsArtifactRepo` needs to interact with tracing service to look up a run / an experiment. Relying on the global tracking URI sometimes often leads to issues.